diff --git a/misc/im.kaidan.kaidan.desktop b/misc/im.kaidan.kaidan.desktop
index 1d641fc5..00ad03c3 100755
--- a/misc/im.kaidan.kaidan.desktop
+++ b/misc/im.kaidan.kaidan.desktop
@@ -100,10 +100,17 @@ Keywords[sv]=chatt;direktmeddelanden;jabber;xmpp;qt;nätverk;
 Keywords[uk]=chat;messaging;im;jabber;xmpp;qt;network;балачка;спілкування;обмін повідомленнями;миттєвий;джаббер;жабер;іксемпепе;к'ют;мережа;
 Keywords[x-test]=xxchatxx;xxmessagingxx;xximxx;xxjabberxx;xxxmppxx;xxqtxx;xxnetworkxx;
 Keywords[zh_TW]=chat;messaging;im;jabber;xmpp;qt;network;聊天;對話;傳訊;訊息;即時;通訊;網路;網絡;
-Exec=kaidan %u
+Exec=qt-runner kaidan %u
 Icon=kaidan
 StartupNotify=true
 StartupWMClass=Kaidan
 Terminal=false
 Categories=Network;InstantMessaging;Qt;
 MimeType=x-scheme-handler/xmpp;
+
+X-Nemo-Single-Instance=no
+X-Nemo-Application-Type=no-invoker
+
+[X-Sailjail]
+Sandboxing=Disabled
+
